### PR TITLE
Set internal metadata ETags for SharePoint

### PR DESCRIPTION
**Pull request by Yosif-Smint**
## Change description

> Description here

## Type of change
- [ ] Bug fix (fixes an issue)
- [ ] New feature (adds functionality)
- [ ] Maintenance (code clean-up, etc.)

## Related issues

> Reference related issues here

### Development

- [ ] Lint rules pass locally
- [ ] Application changes have been tested thoroughly
- [ ] Automated tests covering modified code pass

### Security

- [ ] Security impact of change has been considered
- [ ] Code follows company security practices and guidelines

### Network

- [ ] Changes to network configurations have been reviewed
- [ ] Any newly exposed public endpoints or data have gone through security review

### Code review 

- [ ] Pull request has a descriptive title and context useful to a reviewer. Screenshots or screencasts are attached as necessary
- [ ] "Ready for review" label attached and reviewers assigned
- [ ] Changes have been reviewed by at least one other contributor
- [ ] Pull request is linked to task tracker where applicable

### CODE REVIEW

File: Examples/Backend/DataAdapters/DataAdapter-SharePoint/Assets/Common/SharepointContentConverter.cs [hunk 1]
```diff
@@ -396,6 +409,35 @@ private void SetThumbnails(AssetDataObject assetDataObject, DriveItem driveItem)
             }
         }
 
+        private static string GetETag(string url)
+        {
+            if (string.IsNullOrEmpty(url))
```
Comment (Yosif-Smint) [inline Examples/Backend/DataAdapters/DataAdapter-SharePoint/Assets/Common/SharepointContentConverter.cs:414]: A SharePoint URL looks like this
`https://westeurope1-mediap.svc.ms/transform/thumbnail?provider=spo&inputFormat=jpg&cs=ZGViNDczYTgtOGVlMC00MGM1LWFhNWQtNWQ4MjFkMzA4YzRmfFNQTw&docid=https%3A%2F%2Fsmintio.sharepoint.com%2F_api%2Fv2.0%2Fdrives%2Fb!-avcvthnlUqATJ-UzTU3I3mbry9l6XVEkLlTG4328eeeVVXQ4l-TTaqOAdAoMfB9%2Fitems%2F013NBSTFAI4FRKSFASORALG6TU6HHZOLQA%3Ftempauth%3DeyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJhcHBfZGlzcGxheW5hbWUiOiJTbWludC5pbyBQb3J0YWxzIiwiYXBwaWQiOiJkZWI0NzNhOC04ZWUwLTQwYzUtYWE1ZC01ZDgyMWQzMDhjNGYiLCJhdWQiOiIwMDAwMDAwMy0wMDAwLTBmZjEtY2UwMC0wMDAwMDAwMDAwMDAvc21pbnRpby5zaGFyZXBvaW50LmNvbUBlYzk2YjlhNi0yYTRjLTQxOGEtYjkxYi04YTVjMmRhNTIyOGIiLCJjYWNoZWtleSI6IjBoLmZ8bWVtYmVyc2hpcHwxMDAzMjAwMTk1NDZmNWM1QGxpdmUuY29tIiwiZW5kcG9pbnR1cmwiOiI0OWM5Rzc5S25jR2RCdE9kcVdDN1NUc3p0T0UvQnNuVjVFa1BGU2Q5MEo4PSIsImVuZHBvaW50dXJsTGVuZ3RoIjoiMTU1IiwiZXhwIjoiMTcxNzA4MTIwMCIsImZhbWlseV9uYW1lIjoiVmVsZXYiLCJnaXZlbl9uYW1lIjoiWW9zaWYiLCJpcGFkZHIiOiIyMC4xOTAuMTgzLjI1IiwiaXNsb29wYmFjayI6IlRydWUiLCJpc3MiOiIwMDAwMDAwMy0wMDAwLTBmZjEtY2UwMC0wMDAwMDAwMDAwMDAiLCJuYmYiOiIxNzE3MDU5NjAwIiwicHVpZCI6IjEwMDMyMDAxOTU0NkY1QzUiLCJzY3AiOiJhbGxzaXRlcy5yZWFkIGFsbHNpdGVzLnJlYWQgYWxsc2l0ZXMud3JpdGUgZmFzdFNlYXJjaC5xdWVyeXNwIGFsbHByb2ZpbGVzLnJlYWQiLCJzaWduaW5fc3RhdGUiOiJbXCJrbXNpXCJdIiwic2l0ZWlkIjoiWW1Wa1kyRmlaamt0Tmpka09DMDBZVGsxTFRnd05HTXRPV1k1TkdOa016VXpOekl6IiwidGlkIjoiZWM5NmI5YTYtMmE0Yy00MThhLWI5MWItOGE1YzJkYTUyMjhiIiwidHQiOiIyIiwidXBuIjoieW9zaWZAc21pbnQuaW8iLCJ2ZXIiOiJoYXNoZWRwcm9vZnRva2VuIn0.h2jfoMKdeMykIzQVmAQxBX-C54Qxl3eARnPxX7nUAWQ%26version%3DPublished&width=800&height=800&cb=63852404529`

The query string has an encoded URL from which `https://westeurope1-mediap.svc.ms/transform` reads data
The docId query string is the encoded URL `&docid=https://smintio.sharepoint.com/_api/v2.0/...)` which has its own query parameter `tempauth` which I am removing

The final version of the computed ETAg looks like this
`https://westeurope1-mediap.svc.ms/transform/thumbnail?provider=spo&inputFormat=jpg&cs=ZGViNDczYTgtOGVlMC00MGM1LWFhNWQtNWQ4MjFkMzA4YzRmfFNQTw&docid=https%3a%2f%2fsmintio.sharepoint.com%2f_api%2fv2.0%2fdrives%2fb!-avcvthnlUqATJ-UzTU3I3mbry9l6XVEkLlTG4328eeeVVXQ4l-TTaqOAdAoMfB9%2fitems%2f013NBSTFGS75WULWPEJ5DJ5SKQCPFR7BSU&width=800&height=800&cb=63852404529`

Long story short we are removing the `tempauth` from the encoded URL query parameter

File: Examples/Backend/DataAdapters/DataAdapter-SharePoint/Assets/Common/SharepointContentConverter.cs [hunk 2]
```diff
@@ -333,6 +338,10 @@ private void SetThumbnails(AssetDataObject assetDataObject, DriveItem driveItem)
             {
                 var thumbnailSet = driveItem.Thumbnails?.FirstOrDefault();
 
+                assetDataObject.InternalMetadata.SmallThumbnailETag = GetETag(thumbnailSet?.Small?.Url);
```
Comment (Yosif-Smint) [inline Examples/Backend/DataAdapters/DataAdapter-SharePoint/Assets/Common/SharepointContentConverter.cs:341]: When `IsPlaybackLargeAvailable == true` unfortunately we are using the Graph API and we don't have the URL
The same is true when `IsPdfPreviewAvailable == true` and `IsThumbnailPreviewAvailable == true`

File: Examples/Backend/DataAdapters/DataAdapter-SharePoint/Assets/Common/SharepointContentConverter.cs [hunk 3]
```diff
@@ -324,6 +325,10 @@ private void SetThumbnails(AssetDataObject assetDataObject, DriveItem driveItem)
...
```
Comment (smintrh78) [inline Examples/Backend/DataAdapters/DataAdapter-SharePoint/Assets/Common/SharepointContentConverter.cs:328]: Is it possible in SharePoint to replace previews?

Comment (Yosif-Smint) [reply Examples/Backend/DataAdapters/DataAdapter-SharePoint/Assets/Common/SharepointContentConverter.cs:328]: I couldn't find a way to do this through their UI

Comment (smintrh78) [reply Examples/Backend/DataAdapters/DataAdapter-SharePoint/Assets/Common/SharepointContentConverter.cs:328]: Then we do not need that code
